src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/attachment/v2/AwsV2AttachmentPlugin.java

```diff
@@ -59,87 +55,87 @@ public boolean hasFailed(String instanceState) {
+
+        DetachVolumeRequest request = DetachVolumeRequest.builder()
+                .volumeId(volumeId)
+                .build();
+
+        try {
+            client.detachVolume(request);
+        } catch (Exception e) {
+            LOGGER.error(String.format(Messages.Error.ERROR_WHILE_REMOVING_RESOURCE, RESOURCE_NAME, volumeId), e);
+            throw new UnexpectedException();
+        }
+    }
+
+    @Override
+    public AttachmentInstance getInstance(AttachmentOrder attachmentOrder, AwsV2User cloudUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE_S, attachmentOrder.getInstanceId()));
+
+        Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+        String attachmentId = attachmentOrder.getInstanceId();
+        
+        DescribeVolumesRequest request = DescribeVolumesRequest.builder()
+                .volumeIds(attachmentId)
+                .build();
+        
+        DescribeVolumesResponse response = AwsV2CloudUtil.doDescribeVolumesRequest(client, request);
+        return mountAttachmentInstance(response);
+    }
 
 	protected AttachmentInstance mountAttachmentInstance(DescribeVolumesResponse response)
-			throws InstanceNotFoundException {
+			throws FogbowException {
 
-		VolumeAttachment attachment = getFirstAttachment(response);
+	    Volume volume = AwsV2CloudUtil.getVolumeFrom(response);
+		VolumeAttachment attachment = getAttachmentBy(volume);
 		String cloudState = attachment.stateAsString();
 		String computeId = attachment.instanceId();
 		String volumeId = attachment.volumeId();
 		String device = attachment.device();
 		return new AttachmentInstance(volumeId, cloudState, computeId, volumeId, device);
 	}
 
-	protected VolumeAttachment getFirstAttachment(DescribeVolumesResponse response) throws InstanceNotFoundException {
-		Volume volume = getFirstVolume(response);
+	protected VolumeAttachment getAttachmentBy(Volume volume) throws FogbowException {
 		if (!volume.attachments().isEmpty()) {
-			return volume.attachments().get(FIRST_POSITION);
-		}
-		throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
-	}
-
-	protected Volume getFirstVolume(DescribeVolumesResponse response) throws InstanceNotFoundException {
-		if (!response.volumes().isEmpty()) {
-			return response.volumes().get(FIRST_POSITION);
+			return volume.attachments().listIterator().next();
 		}
 		throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
 	}
+	
+	private String doAttachVolumeRequest(Ec2Client client, AttachVolumeRequest request)
+            throws FogbowException {
+
+        String attachmentId;
+        try {
+            client.attachVolume(request);
+            attachmentId = request.volumeId();
+        } catch (Exception e) {
+            throw new UnexpectedException(String.format(Messages.Exception.GENERIC_EXCEPTION, e), e);
+        }
+        return attachmentId;
+    }
```

Comment:
please, add an internal comment in this defineDeviceNameAttached method. it is not obvious why we need this

Reply:
@wassoares I think you can do that better than me...

Reply:
Okay, I included comments on the method as requested and made corrections to it.